Src/FluentAssertions/Equivalency/MemberVisibilityExtensions.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Concurrent;
+using Reflectify;
+
+namespace FluentAssertions.Equivalency;
+
+internal static class MemberVisibilityExtensions
+{
+    private static readonly ConcurrentDictionary<MemberVisibility, MemberKind> Cache = new();
+
+    public static MemberKind ToMemberKind(this MemberVisibility visibility)
+    {
+        return Cache.GetOrAdd(visibility, v =>
+        {
+            MemberKind result = MemberKind.None;
+
+            foreach (MemberVisibility flag in Enum.GetValues(typeof(MemberVisibility)))
```

Comment:
We could polyfill `Enum.GetValues<T>` for older frameworks like we've done for other APIs in e.g. `SystemExtensions` and `StringBuilderExtensions` that PolySharp doesn't already provide.

| Method     | Job                  | Runtime              | Mean      | Error     | StdDev    | Ratio | RatioSD | Gen0   | Allocated | Alloc Ratio |
|----------- |--------------------- |--------------------- |----------:|----------:|----------:|------:|--------:|-------:|----------:|------------:|
| NonGeneric | .NET 8.0             | .NET 8.0             | 101.05 ns |  0.896 ns |  0.794 ns |  1.00 |    0.01 | 0.0076 |      48 B |        1.00 |
| Generic    | .NET 8.0             | .NET 8.0             |  32.72 ns |  0.272 ns |  0.241 ns |  0.32 |    0.00 | 0.0076 |      48 B |        1.00 |
|            |                      |                      |           |           |           |       |         |        |           |             |
| NonGeneric | .NET Framework 4.7.2 | .NET Framework 4.7.2 | 744.11 ns | 14.475 ns | 15.488 ns |  1.00 |    0.03 | 0.0267 |     169 B |        1.00 |
| Generic    | .NET Framework 4.7.2 | .NET Framework 4.7.2 | 738.11 ns |  3.947 ns |  3.692 ns |  0.99 |    0.02 | 0.0267 |     169 B |        1.00 |

```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Jobs;
using BenchmarkDotNet.Running;
using System;

BenchmarkRunner.Run<GetValuesBenchmark>();

[MemoryDiagnoser]
[SimpleJob(RuntimeMoniker.Net472)]
[SimpleJob(RuntimeMoniker.Net80)]
public class GetValuesBenchmark
{
    [Benchmark(Baseline = true)]
    public MemberVisibility[] NonGeneric() => (MemberVisibility[])Enum.GetValues(typeof(MemberVisibility));

    [Benchmark]
#if NET6_0_OR_GREATER
    public MemberVisibility[] Generic() => Enum.GetValues<MemberVisibility>();
#else
    public MemberVisibility[] Generic() => (MemberVisibility[])Enum.GetValues(typeof(MemberVisibility));
#endif
}

[Flags]
public enum MemberVisibility
{
    None = 0,
    Internal = 1,
    Public = 2,
    ExplicitlyImplemented = 4,
    DefaultInterfaceProperties = 8
}
```

Reply:
I just used an #ifdef for now.